packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -220,36 +220,49 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
         global.accounts.push(...accounts);
       });
 
-      beforeEach(async function () {
-        const basicPlans = await hbarSpendingPlanRepository.findAllActiveBySubscriptionTier(
-          [SubscriptionTier.BASIC],
-          requestDetails,
-        );
-        for (const plan of basicPlans) {
-          await hbarSpendingPlanRepository.delete(plan.id, requestDetails);
-          await ethAddressSpendingPlanRepository.deleteAllByPlanId(plan.id, 'before', requestDetails);
-          await ipSpendingPlanRepository.deleteAllByPlanId(plan.id, 'before', requestDetails);
-        }
+      afterEach(async () => {
+        hbarSpendingPlanRepository.resetAmountSpentOfAllPlans(requestDetails);
+
+        // Note: Since the total HBAR budget is shared across the entire Relay instance by multiple test cases,
+        //       and expense updates occur asynchronously, the wait below ensures that the HBAR amount has sufficient time
+        //       to update properly after each test.
+        await Utils.wait(1500);
       });
 
-      describe('Total HBAR Limit', () => {
-        it('should execute "eth_sendRawTransaction" without triggering HBAR rate limit exceeded', async function () {
-          const parentContract = await deployContract(parentContractJson, accounts[0].wallet);
+      describe('@hbarlimiter-batch1 Total HBAR Limit', () => {
+        const pollForProperRemainingHbar = async (initialRemainingHbars: number, expectedTxCost: number) => {
+          let updatedRemainingHbarsAfter = Number(await metrics.get(testConstants.METRICS.REMAINING_HBAR_LIMIT));
 
-          // awaiting for HBAR limiter to finish updating expenses in the background
-          await Utils.wait(6000);
+          // Note: expectedTxCost may be retrieved from mirror node which doesn't include the getRecord transaction fee.
+          //       calculating delta = expectedTxCost * tolerance to account for this difference in transaction costs.
+          const delta = expectedTxCost * 0.002;
 
-          const parentContractAddress = parentContract.target as string;
-          global.logger.trace(
-            `${requestDetails.formattedRequestId} Deploy parent contract on address ${parentContractAddress}`,
+          while (initialRemainingHbars - updatedRemainingHbarsAfter > expectedTxCost + delta) {
```

Comment:
**Note for reviewer:** 
- `initialRemainingHbars`: the remaining HBARs retrieved at the begining of the test
- `updatedRemainingHbarsAfter`: the remaining HBARs retrieved AFTER making calls to the network
- `expectedTxCost`: expected total transacction cost retrieved from mirror node
- `delta`: `expectedTxCost` may be retrieved from mirror node which doesn't include the transaction fee from getRecord queries. Delta is calculated to account for this difference in transaction costs.

Ideally, after a call to the network, `initialRemainingHbars - updatedRemainingHbarsAfter` should be greater than `expectedTxCost + delta`. However, the test client sometimes fails to retrieve the correct amounts, meaning `updatedRemainingHbarsAfter` could be fetched slightly too early, before the total burnt HBAR amount has fully updated in the metrics. In such cases, continue polling until the ideal amount is retrieved.